overlord/devicestate/handlers_install.go

```diff
@@ -250,6 +250,10 @@ func writeTimings(st *state.State, rootdir, fromMode string) error {
 	return nil
 }
 
+func (m *DeviceManager) doPrepareUbuntuSave(t *state.Task, _ *tomb.Tomb) error {
+	return m.maybeSetupUbuntuSave()
+}
```

Comment:
wondering whether it would make sense to merge this into doSetupRunSystem ?

Reply:
This was actually my initial approach, to just add it as a step after the installRun in doSetupRunSystem. However it felt very 'patchy' (also because it doesn't always need to run) and the code felt out of place. I know we don't just create new tasks if it can be avoided, but it felt much cleaner in this case. If you feel strong about merging it in, I will